client/material/text-field.tsx

```diff
@@ -212,6 +218,12 @@ export const TextField = React.forwardRef<HTMLInputElement, TextFieldProps>(
       [dense, rows],
     )
 
+    useEffect(() => {
```

Comment:
You want this in a `useLayoutEffect` so the resize happens before the browser paints, otherwise you'll get a flash of the smaller size